### PR TITLE
Emails: enable the button to add emails so we rely on messaging in em…

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
#### Proposed Changes

This change will have the "Add Email" button always active. If there is some reason why the user is not able to add an email, then we will rely on the information displayed in the Email Comparison screen.

I've also taken the opportunity to convert the React Class Component into a functional react in TypeScript.

In order to ease the review, the main change was to remove a flag that was disabling / not rendering the button to add email, the referenced flag is `canCurrentUserAddEmail( domain )`
```
{ canCurrentUserAddEmail( domain ) && (
       <Button
 	      href={ emailManagementPurchaseNewEmailAccount(
 		      selectedSiteSlug,
 		      domain.name,
 		      currentRoute,
 		      source
 	      ) }
       >
 	      { translate( 'Add Email' ) }
       </Button>
) }
```

So newly created function component in TypeScript, will no longer have the ` canCurrentUserAddEmail( domain )` check

#### Testing Instructions

1. Have a domain that does not belong to you in your site (add a user to your site, and transfer the domain within calypso)
2. Assert that you can add email for that domain and you see the message about why you can't add an email to the domain.

*

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

**Comment by fredrikekelund** [comment]
This is a duplicate of #68156

**Comment by fredrikekelund** [comment]
Sorry, I stand corrected! The change to always display the `Add Email` buttons in `EmailListInactive` is already covered in #68156. But the warning message in `EmailPlanWarnings` isn't.

File: client/my-sites/email/email-management/home/email-plan-warnings.tsx [hunk 1]
```diff
@@ -39,6 +39,13 @@ const EmailPlanWarnings = ( { domain, emailAccount }: EmailPlanWarningsProps ) =
 	const cannotAddEmailWarningCode = cannotAddEmailWarningReason?.code ?? null;
 
 	const WarningMessageForCode = ( props: { code: string | number | null } ) => {
+		return (
```
Comment (wongasy) [inline client/my-sites/email/email-management/home/email-plan-warnings.tsx:42]: I suspect this is test code?

**Comment by wongasy** [comment]
@AllTerrainDeveloper As @fredrikekelund pointed out, the button has already been handled in https://github.com/Automattic/wp-calypso/pull/68156. We've been holding off deploying that PR, since we're waiting for https://github.com/Automattic/wp-calypso/pull/68158 to be completed.

**Comment by fredrikekelund** [comment]
@AllTerrainDeveloper I guess we should close this?

**Pull request closed without merging**